window.c

```diff
@@ -68,10 +79,34 @@ void vtk_window_mainloop(vtk_window win) {
 	XSelectInput(win->root->dpy, win->w, win->event_mask);
 	XMapWindow(win->root->dpy, win->w);
 
+	int x11_fd = ConnectionNumber(win->root->dpy);
+
+	fd_set fds;
+
+	int nfds = win->updatefd;
+	if (x11_fd > nfds) nfds = x11_fd;
+	++nfds;
+
 	XEvent ev;
 	while (!win->should_close) {
-		XNextEvent(win->root->dpy, &ev);
-		vtk_event_handle(win, ev);
+		FD_ZERO(&fds);
+		FD_SET(x11_fd, &fds);
+		FD_SET(win->updatefd, &fds);
+		select(nfds, &fds, NULL, NULL, NULL);
```

Comment:
`select` -> `poll`

select is slower, more annoying to use, and only works on fds < 1024
epoll is even better, but Linux-specific and it'd be nice if vtk works in other Unix-like environments, even if it's not really supported